### PR TITLE
Protected divisions to get rid of crashes on CSC/Sisu with Vlasiator.

### PR DESCRIPTION
**Pull request by ykempf**
This set of modifications seem to have got me rid of the crashes we were experiencing in recent days on CSC/Sisu with Vlasiator.

### CODE REVIEW

**Comment by galfthan** [comment]
It should be, one reason why I did not protect them originally. Thanks for contribution, I'll review and merge.

**Comment by ykempf** [comment]
But then I might not have fixed the bug indeed... I was getting bad_alloc and the like in phiporf::stop in omp parallel regions, which is not touched here. So it is still a Heisenbug maybe.

**Comment by ykempf** [comment]
The last commit is not tested yet but one of the two divisions has caused an arithmetic error in Vlasiator. I'll confirm here once the test gets through in a few hours that it is fixed.

**Comment by ykempf** [comment]
The test ran through now, I can confirm that the arithmetic error we saw is fixed by the latest commit.

File: src/paralleltimertree.cpp [hunk 1]
```diff
@@ -422,16 +425,31 @@ bool ParallelTimerTree::printTimers(double minFraction, const std::map<std::stri
...
+            if(nProcessesInPrint>0)
+               table.addElement(stats.timeSum[i]/nProcessesInPrint);
+            else
+               table.addElement(0.0);
+            
             table.addElement(100.0 * stats.timeParentFraction[i]);
-            double imbTime = stats.timeMax[i].val - stats.timeSum[i]/nProcessesInPrint;            
-            if(nProcessesInPrint>1)
+            
+            if(nProcessesInPrint>0 && stats.timeMax[i].val != 0.0) {
```
Comment (galfthan) [inline src/paralleltimertree.cpp]: should be >1

File: src/paralleltimertree.cpp [hunk 2]
```diff
@@ -422,16 +425,31 @@ bool ParallelTimerTree::printTimers(double minFraction, const std::map<std::stri
...
+               double imbTime = stats.timeMax[i].val - stats.timeSum[i]/nProcessesInPrint;
                table.addElement(100 * imbTime / stats.timeMax[i].val  * ( nProcessesInPrint /(nProcessesInPrint - 1 )));
+            }
+            else{
+               table.addElement(0.0);
+            }
+
+            if(nProcessesInPrint>0)
```
Comment (galfthan) [inline src/paralleltimertree.cpp]: I have tried to have spaces in the expressions for readability reasons:  (nProcessesInPrint > 0)

**Comment by galfthan** [comment]
This is in

**Pull request closed without merging**